common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Image.scala

```diff
@@ -30,8 +30,39 @@ case class MergedImage[Id <: IdState](
       id = id,
       location = location
     )
+
+  def augment(inferredData: => InferredData): AugmentedImage[Id] =
+    augmentWithOption(Some(inferredData))
+
+  def augmentWithNone: AugmentedImage[Id] = augmentWithOption(None)
+
+  private def augmentWithOption(
+    inferredData: => Option[InferredData]): AugmentedImage[Id] =
+    AugmentedImage[Id](
+      id = id,
+      location = location,
+      parentWork = parentWork,
+      fullText = fullText,
+      inferredData = inferredData
+    )
 }
 
+case class AugmentedImage[Id <: IdState](
+  id: Id,
+  location: DigitalLocation,
+  parentWork: Id,
+  fullText: Option[String] = None,
+  inferredData: Option[InferredData] = None
+) extends BaseImage[Id]
+
+case class InferredData(
+  // We split the feature vector so that it can fit into
+  // ES's dense vector type (max length 2048)
+  features1: List[Float],
```

Comment:
Would it be worth making this a 2048 sized list type?

Reply:
Maybe a `Vector[Float]`?

Reply:
I can't find a sized list type, where does it exist?

Reply:
There isn't size list afaik (thats an `Array`), but from your comment it is "max size" rather than "size"?

Reply:
The ES limitation is a max but this data is a definite size. I can change it to a `Vector` but I'm not sure what advantages we gain from this other than a more fitting name...

Reply:
List docs:

> This class is optimal for last-in-first-out (LIFO), stack-like access patterns. If you need another access pattern, for example, random access or FIFO, consider using a collection more suited to this than List.

Vector docs:

> Vector is a general-purpose, immutable data structure. It provides random access and updates in effectively constant time, as well as very fast append and prepend. Because vectors strike a good balance between fast random selections and fast random functional updates, they are currently the default implementation of immutable indexed sequences.

Reply:
Exactly - we never access these List elements except for mapping and slicing

Reply:
Ah ok, I get you. Might still use vector anyway because of the name, and the fact that it shows intent of it being a indexable data structure but :man_shrugging:

Reply:
Yea, you can fix the size of an array, but this is a max size.
Would it just be a matter of having something like:
```
case class DenseVector(features: List[Float]) {
  def apply(features: List[Float]) = {
    if (features.length) { /* blow up */ }
  }
}
```

Just so that we can error on that instead of erroring on ingest

Reply:
You could probably have that case class have a list of lists that never exceed 2048 negating the need for the `features1` and `features2` and just splits accordingly. The index can also allow for a dynamic set of the features, but not sure if that will make it hard to use.

Reply:
I don't think there would ever be a dynamic set of feature vectors; they're being stored in the index for the convenience of the model builder (rather than as something to query).

I'll add a check on size in `augmentInput`, but whilst we can't express the fixed size in the type system, I'd rather not go full defensive programming on it. We trust the inferrer, it's coupled not only to the inference manager but to the `InferredData` itself, and it [does contain](https://github.com/wellcomecollection/data-science/blob/master/apis/feature_extraction/inference/app/src/feature_extraction.py) the guarantee on the size of this list.